src/controllers/contracts/ContractsInkController.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { ApiPromise } from '@polkadot/api';
+import { ContractPromise } from '@polkadot/api-contract';
+import { RequestHandler } from 'express';
+import { BadRequest } from 'http-errors';
+
+import { validateAddress } from '../../middleware';
+import { ContractsInkService } from '../../services';
+import {
+	IBodyContractMetadata,
+	IContractQueryParam,
+	IPostRequestHandler,
+} from '../../types/requests';
+import AbstractController from '../AbstractController';
+
+export default class ContractsInkController extends AbstractController<ContractsInkService> {
+	constructor(api: ApiPromise) {
+		super(api, '/contracts/ink/:address', new ContractsInkService(api));
+		this.initRoutes();
+	}
+
+	protected initRoutes(): void {
+		this.router.use(this.path, validateAddress);
+		this.safeMountAsyncPostHandlers([
+			['/query', this.callContractQuery as RequestHandler],
+		]);
+	}
+
+	/**
+	 * Send a message call to a contract. It defaults to get if nothing is inputted.
+	 *
+	 * @param _req
+	 * @param res
+	 */
+	private callContractQuery: IPostRequestHandler<
+		IBodyContractMetadata,
+		IContractQueryParam
+	> = async (
+		{
+			params: { address },
+			body,
+			query: { method = 'get', gasLimit, storageDepositLimit, args },
```

Comment:
GET by default but containing a body I think is techinically ok, but I'm hesitant!

https://stackoverflow.com/questions/978061/http-get-with-request-body was an interesting read, and it points out that many caching things and layers between eg sidecar the the world might ignore bodies in GET requests and could hand back cached results or something.

This is just a default though, and the openapi bit says POST, so maybe just set this default to POST too?

Reply:
Ahh So i think you are misunderstanding what the `get` method is for. This is the method that will be used to call the contract. `get` just happens to be the default standard. 

That being said the endpoint just uses a POST request to attach the body and receive a response. After some research it seemed like the appropriate decision.

Reply:
Ooohhh I see! Ok that makes much more sense, thanks!